### PR TITLE
Stac 12968 centralized auth config (v4.4)

### PR DESCRIPTION
**Pull request by konstantintatarnikov**
- Centralized authorization config
- upgrade notes

### CODE REVIEW

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 1]
```diff
@@ -18,13 +18,99 @@ This page provides specific instructions for upgrading to each currently support
...
+   ```
+   For details you can consult with the [Default and custom role names](../../configure/security/rbac/rbac_permissions.md#default-and-custom-role-names) section.
+
+   If you are still not sure what you need to do, please contact [StackState support](https://support.stackstate.com/hc/en-us)
 {% endtab %}
 
 {% tab title="Linux" %}
 #### v4.4.0
+
+* Authorization is configured in single place for Base and Admin Api.
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:71]: * Authorization is now configured in single place for Base API and Admin API.

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 2]
```diff
@@ -18,13 +18,99 @@ This page provides specific instructions for upgrading to each currently support
...
+        adminGroups = ["your-custom-oidc-or-ldap-or-keycloak-admin-role"]
+        powerUserGroups = ["your-custom-oidc-or-ldap-or-keycloak-power-user-role"]
+        guestGroups = ["your-custom-oidc-or-ldap-or-keycloak-guest-role"]
+        ...
+      }
+    }
+  }
+  ```
+
+  You have to move subject-role mappings to centralized authorization configuration, as in example below.
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:89]: The subject-role mappings must be moved to a centralized authorization configuration using the syntax `xxxGroups = ${stackstate.authorization.xxxGroups} ["custom-role"]` as shown in the example below.

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 3]
```diff
@@ -18,13 +18,99 @@ This page provides specific instructions for upgrading to each currently support
...
+        // no subject-role mappings here
+        ...
+      }
+    }
+  }
+  ```
+
+  Please note that you have to use syntax:
+  `xxxGroups = ${stackstate.authorization.xxxGroups} ["your-custom-role"]`
+  to extend the existing lists that contain default roles `stackstate-admin`, `stackstate-guest` and `stackstate-power-user`.
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:110]: {% hint style="info" %}
The list of roles will be extended to include the new, custom roles. The default roles will remain available (stackstate-admin, stackstate-guest and stackstate-power-user).
{% endhint %}

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 4]
```diff
@@ -18,13 +18,99 @@ This page provides specific instructions for upgrading to each currently support
...
+
+  If you are still not sure what you need to do, please contact [StackState support](https://support.stackstate.com/hc/en-us)
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:112]: If you are still not sure what you need to do, contact [StackState support](https://support.stackstate.com/hc/en-us).

File: configure/security/rbac/rbac_permissions.md [hunk 5]
```diff
@@ -41,11 +41,11 @@ StackState comes with three predefined roles:
 
 ### Default and custom role names
 
-The default names for the pre-defined roles \(`stackstate-admin`, `stackstate-power-user`, `stackstate-guest`\) can be overridden. In the same way, extra roles can also be added that have the same permissions. Below is an example of how to do this for both Kubernetes and Linux installations.
+The default pre-defined roles names \(`stackstate-admin`, `stackstate-power-user`, `stackstate-guest`\) are always available. In the same time, extra roles can also be added that have the same permissions. Below is an example of how to do this for both Kubernetes and Linux installations.
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply configure/security/rbac/rbac_permissions.md:44]: fixed

File: configure/security/authentication/keycloak.md [hunk 6]
```diff
@@ -91,6 +91,18 @@ To configure StackState to use a KeyCloak authentication provider on Linux, KeyC
 {% tabs %}
 {% tab title="application\_stackstate.conf" %}
 ```javascript
+authorization {
+  // map the roles from Keycloak to the
+  // 3 standard subjects in StackState (guest, powerUser and admin)
+  // Please note! you have to use syntax
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply configure/security/authentication/keycloak.md:97]: fixed

File: configure/security/authentication/oidc.md [hunk 7]
```diff
@@ -87,6 +87,19 @@ To configure StackState to use an OIDC authentication provider on Linux, OIDC de
 {% tabs %}
 {% tab title="application\_stackstate.conf" %}
 ```javascript
+
+authorization {
+  // map the groups from the OIDC provider to the
+  // 3 standard subjects in StackState (guestGroups, powerUserGroups and adminGroups)
+  // Please note! you have to use syntax
+  // `<group>Groups = ${stackstate.authorization.<group>Groups} ["oidc-role"]`
+  // to extend the list of standard subjects (stackstate-admin, stackstate-guest, stackstate-power-user)
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply configure/security/authentication/oidc.md:96]: fixed

File: configure/security/rbac/rbac_permissions.md [hunk 8]
```diff
@@ -72,16 +70,18 @@ stackstate/stackstate
 {% endtab %}
 
 {% tab title="Linux" %}
-Edit the existing keys in the `authentication` section \(nested in `stackstate.api`\) in the configuration file like in the example to replace the default role names with these custom role names. Restart StackState to make the change take effect.
+To extend the default role names with custom role names, you should edit the existing keys in the `authorization` section in the configuration file like in the example below. Restart StackState to make the change take effect.
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply configure/security/rbac/rbac_permissions.md:74]: Changed, and also I've added `application_context.conf`

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md (hunk 4 above)
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md]: fixed

File: configure/security/rbac/rbac_permissions.md [hunk 9]
```diff
@@ -72,16 +70,18 @@ stackstate/stackstate
...
+authorization {
+  guestGroups = ${stackstate.authorization.guestGroups} ["custom-guest-role"]
+  powerUserGroups = ${stackstate.authorization.powerUserGroups} ["custom-power-user-role"]
+  adminGroups = ${stackstate.authorization.adminGroups} ["custom-admin-role"]
 }
 ```
 
-Of course it is also possible to leave the defaults in place, for example the `guestGroups` would then have an array with 2 entries: `["stackstate-guest", "custom-guest-role"]`.
+The syntax `xxxGroups = ${stackstate.authorization.xxxGroups} ["custom-role"]` allows extending the existing lists and therefore the default roles `stackstate-admin`, `stackstate-guest` and `stackstate-power-user` remain available.
+
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply configure/security/rbac/rbac_permissions.md:88]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md (hunk 1 above)
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 10]
```diff
@@ -18,13 +18,99 @@ This page provides specific instructions for upgrading to each currently support
 
 ### Upgrade to v4.4.x
 
+* Baselines have been disabled in v4.4. The `BaselineFunction` and `Baseline` objects are still available, but they do not serve any purpose other than smooth transition to Autonomous Anomaly Detector (AAD) framework. If you have custom StackPacks that auto-create baselines this is the last moment to remove baselines from templates and make transition to AAD. In release v4.5 baselines will be removed completely and templates using them will break.
+* Authorization configuration is centralized now for Base and Admin Api. This means that there is single location in the configuration for groups to roles mappings with 3 default roles that previously could be overridden.
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 11]
```diff
@@ -18,13 +18,99 @@ This page provides specific instructions for upgrading to each currently support
...
+```
+This has impact on stackstate upgrades if you have overrides the authentication config. Please see corresponding kubernetes or linux section below.
+
 {% tabs %}
 {% tab title="Kubernetes" %}
 #### v4.4.0
+
+* Authorization is configured in single place for Base and Admin Api.
+
+  In general case you don't have to do any changes unless you have configured api role overrides for specific services.
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md:40]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 12]
```diff
@@ -18,13 +18,99 @@ This page provides specific instructions for upgrading to each currently support
 
 ### Upgrade to v4.4.x
 
+* Baselines have been disabled in v4.4. The `BaselineFunction` and `Baseline` objects are still available, but they do not serve any purpose other than smooth transition to Autonomous Anomaly Detector (AAD) framework. If you have custom StackPacks that auto-create baselines this is the last moment to remove baselines from templates and make transition to AAD. In release v4.5 baselines will be removed completely and templates using them will break.
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md:21]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 13]
```diff
@@ -18,13 +18,99 @@ This page provides specific instructions for upgrading to each currently support
 
 ### Upgrade to v4.4.x
```
[the remark this answers is not in the document]
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md]: I've duplicated the upgrade info in k8s and linux sections. it seems less complex for the user.

**Pull request merged**